src/media/js/compat_filter.js

```diff
@@ -0,0 +1,147 @@
+/*
+    Compatibility filtering and device filtering. Mainly used to add arguments
+    to the API router's processor. It also handles `limit` since that can be
+    affected by the device or form factor.
+
+    Device filtering is affected by a dropdown found above app lists. When the
+    dropdown is changed, we will persist the new device filtering preference
+    globally and across sessions.
+
+    For desktop, device filtering is off by default at the moment.
+
+    For feature profiles, if not in querystring, we default to a feature
+    profile sig that is specially crafted.  Includes all features except more
+    recent ones to hide apps using more recent features from older Marketplaces
+    that have not been updated yet. Generated in Zamboni with:
+        f = FeatureProfile.from_signature('fffffffffffffffffffffffffff')
+        f.update({'mobileid': False, 'precompile_asmjs': False,
+                  'hardware_512mb_ram': False, 'hardware_1gb_ram': False})
+        f.to_signature()
+*/
+define('compat_filter',
+    ['core/capabilities', 'core/l10n', 'core/log', 'core/storage',
+     'core/utils', 'core/views', 'core/z', 'underscore'],
+    function(caps, l10n, log, storage,
+             utils, views, z, _) {
+    'use strict';
+    var logger = log('compat_filter');
+    var gettext = l10n.gettext;
+
+    var limit = 24;
+
+    // Endpoints where feat. profile enabled if conditions met. See apiArgs().
+    var ENDPOINTS_WITH_FEATURE_PROFILE = [
+        'category', 'feed', 'feed-app', 'feed-brand', 'feed-collection',
+        'feed-items', 'feed-shelf', 'installed', 'recommended', 'search'
+    ];
+    var feature_profile = utils.getVars().pro || '7fffffffffff0.51.6';
+
+    var DEVICE_CHOICES = {
+        'android-mobile': gettext('Android Mobile'),
+        'android-tablet': gettext('Android Tablet'),
+        'desktop': gettext('Desktop'),
+        'firefoxos': gettext('Firefox OS'),
+    };
+    var DEVICE_FILTER_CHOICES = [
+        ['all', gettext('All Apps')],
+        ['desktop', gettext('Desktop Apps')],
+        ['firefoxos', gettext('Firefox OS Apps')],
+        ['android-mobile', gettext('Android Mobile Apps')],
+        ['android-tablet', gettext('Android Tablet Apps')]
+    ];
+    var actual_platform = caps.device_platform();
+    var actual_formfactor = caps.device_formfactor();
+    var filterDevice = caps.device_type();
```

Comment:
If you are renaming variables to camelCase, then maybe also take care of the others ?